linera-client/src/config.rs

```diff
@@ -104,19 +106,34 @@ impl<W: Persist<Target = Wallet>> Extend<UserChain> for WalletState<W> {
     }
 }
 
+#[cfg(feature = "fs")]
 impl WalletState<persistent::File<Wallet>> {
-    pub fn create(path: &Path, wallet: Wallet) -> Result<Self, anyhow::Error> {
+    pub fn create_from_file(path: &std::path::Path, wallet: Wallet) -> Result<Self, anyhow::Error> {
         Ok(Self::new(persistent::File::read_or_create(path, || {
             Ok(wallet)
         })?))
     }
 
-    pub fn from_file(path: &Path) -> Result<Self, anyhow::Error> {
+    pub fn read_from_file(path: &std::path::Path) -> Result<Self, anyhow::Error> {
         Ok(Self::new(persistent::File::read(path)?))
     }
 }
 
-impl<W: Persist<Target = Wallet>> WalletState<W> {
+#[cfg(web)]
+impl WalletState<persistent::LocalStorage<Wallet>> {
+    pub fn create_from_local_storage(key: &str, wallet: Wallet) -> Result<Self, anyhow::Error> {
+        Ok(Self::new(persistent::LocalStorage::read_or_create(
+            key,
+            || Ok(wallet),
+        )?))
+    }
+
+    pub fn read_from_local_storage(key: &str) -> Result<Self, anyhow::Error> {
+        Ok(Self::new(persistent::LocalStorage::read(key)?))
+    }
+}
+
+impl<W: std::ops::Deref<Target = Wallet>> WalletState<W> {
```

Comment:
(I think it's fine to import `Deref`.)

Reply:
Oh sorry, I read your suggestion above backwards.  I need to make some changes anyway for tests so I'll include this :)